docs/current_docs/quickstart/basics/basics.mdx

```diff
@@ -0,0 +1,436 @@
+
+```shell
+container | from alpine | with-new-file /hi.txt "Hello from Dagger!" | terminal
+```
+
+This drops you into an interactive terminal running the `bash` shell. You can now directly execute commands in the running container, as shown below:
+
+[FIXME recording]
+
+:::tip INTERACTIVE CONTAINER DEBUGGING
+The `terminal` function is very useful for debugging and experimenting, since it allows you to interact directly with containers and inspect their state, at any stage of your Dagger Function execution.
+:::
+
+## Chain functions
+
+What you just did (and have been doing since the start) is chaining one Dagger API function call to another with the pipe (`|`) operator. This is one of Dagger's most powerful features, as it allows you create dynamic workflows in a single command - no context switching between Dockerfile creation, build commands, and registry pushes.
+
+Dagger's documentation has numerous examples of function chaining in action but here's one more: creating an Alpine container, dropping in a text file with a custom message, setting it to display that message when run, and publishing it to a temporary registry - all in a single command!
+
+```shell
+container | from alpine | with-new-file /hi.txt "Hello from Dagger!" |
+  with-entrypoint cat /hi.txt | publish ttl.sh/hello
+```
+
+[FIXME recording]
+
+
+## Writing custom functions
+
+As your workflows become more complex, you'll start wishing you could make them more reusable, repeatable and shareable. To do this, encapsulate your workflows into custom Dagger Functions. These are just regular code, written in a programming language using a type-safe Dagger SDK and packaged into modules.
+
+Here's the previous example, rewritten as a Dagger Function:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go
+func (m *Basics) Publish(ctx context.Context) (string, error) {
+	return dag.Container().
+		From("alpine:latest").
+		WithNewFile("/hi.txt", "Hello from Dagger!").
+		WithEntrypoint([]string{"cat", "/hi.txt"}).
+		Publish(ctx, "ttl.sh/hello")
+}
+```
+
+</TabItem>
+<TabItem value="Python">
+
+FIXME
+
+</TabItem>
+<TabItem value="TypeScript">
+
+FIXME
+
+</TabItem>
+<TabItem value="PHP">
+
+FIXME
+
+</TabItem>
+<TabItem value="Java">
+
+FIXME
+
+</TabItem>
+</Tabs>
+
+When you create a custom Dagger Function and tell Dagger about it (by installing the corresponding module), the Dagger API is dynamically extended to include that new function. You can then call this function from Dagger Shell, or from your system shell using `dagger call`, in exactly the same way as you would call the original Dagger API.
+
+[FIXME recording]
+
+:::tip FUNCTION CHAINING
+Function chaining works the same way, whether you're writing Dagger Function code using a Dagger SDK or using Dagger Shell. The following are equivalent:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go
+type Basics struct{}
```

Comment:
Is this whole snippet supposed to be in the `:::tip` box?

Reply:
It was, but I've moved it now